clang/test/CIR/IR/invalid.cir

```diff
@@ -173,6 +173,147 @@ cir.func @cast6(%p: !cir.bool) {
 
 // -----
 
+!u32i = !cir.int<u, 32>
+cir.func @cast7(%p: !cir.ptr<!u32i>) {
+  %2 = cir.cast(ptr_to_bool, %p : !cir.ptr<!u32i>), !u32i // expected-error {{requires !cir.bool type for result}}
+  cir.return
+}
+
+// -----
+
+!u32i = !cir.int<u, 32>
+cir.func @cast8(%p: !u32i) {
+  %2 = cir.cast(ptr_to_bool, %p : !u32i), !cir.bool // expected-error {{requires pointer type for source}}
+  cir.return
+}
+
+// -----
+
+!u32i = !cir.int<u, 32>
+cir.func @cast9(%p : !u32i) {
+  %2 = cir.cast(integral, %p : !u32i), f32 // expected-error {{requires !IntegerType for result}}
```

Comment:
Looks like you also found a bug here, this should print `!cir.int`, I wonder if we are still using `mlir::IntegerType` instead of `cir::IntType` or if the error message is just not ideal. Let's track this on another issue.

Reply:
https://github.com/llvm/clangir/issues/318